Source/WebCore/rendering/svg/RenderSVGShape.cpp

```diff
@@ -149,6 +147,7 @@ void RenderSVGShape::layout()
     if (m_needsShapeUpdate) {
         // FIXME: [LBSE] Upstream SVGLengthContext changes
         // graphicsElement().updateLengthContext();
+        m_shapeType = ShapeType::Empty;
```

Comment:
Yeah, let's put this in derived class' `updateShapeFromElement()`. So we can clear this state first to empty, and using early return when it is actually empty (for example, empty rect needs to be empty state).
We do not want to put `Empty` in `LegacyRenderSVGShape::updateShapeFromElement` since it is called from the derived class, and it clears the type to `Empty` incorrectly if we do that. Probably we can remove `LegacyRenderSVGShape::updateShapeFromElement` after strokeBoundingBox computation is purged from that (we will do it lazily to avoid that when it is not necessary, as the same to path materialization), so `LegacyRenderSVGShape::updateShapeFromElement` will become just `m_fillBoundingBox = ensurePath().boundingRect()`. Changed.